### PR TITLE
feat: add select leverage token

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds select token modals to leverage UI
* Converts select token modal to tailwind css
* Adds dark mode
* Adds config option to not show balances (used for eth/btc selector)

## **Test Data or Screenshots**
Bildschirmfoto 2024-04-16 um 10 00 01

Bildschirmfoto 2024-04-16 um 10 00 06

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/leverage/provider.tsx [hunk 1]
```diff
@@ -142,15 +152,33 @@ export function LeverageProvider(props: { children: any }) {
...
+      (token) => token.symbol === tokenSymbol,
+    )[0]
+    if (!token) return
+    setInputToken(token)
+  }
+
+  const onSelectIndexToken = (tokenSymbol: string) => {
+    const token = indexTokens.filter((token) => token.symbol === tokenSymbol)[0]
+    if (!token) return
+    setOutputToken(token)
```
Comment (janndriessen) [inline src/app/leverage/provider.tsx:166]: Will add more logic here later to take into account `isBuying`. Same for above selection function.

File: package.json [hunk 2]
```diff
@@ -21,11 +21,11 @@
     "@emotion/react": "^11.11.0",
     "@emotion/styled": "^11.11.0",
     "@fontsource/open-sauce-sans": "^5.0.7",
-    "@headlessui/react": "1.7.18",
+    "@headlessui/react": "^1.7.19",
```
Comment (0xonramp) [inline package.json:24]: curious - why add the range?

File: src/app/leverage/provider.tsx [hunk 3]
```diff
@@ -142,15 +152,33 @@ export function LeverageProvider(props: { children: any }) {
     setLeverageType(type)
   }
 
+  const onSelectCurrencyToken = (tokenSymbol: string) => {
+    const token = currencyTokens.filter(
+      (token) => token.symbol === tokenSymbol,
+    )[0]
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:158]: nit: can use `find` instead here (and below on line 164)

**Review by 0xonramp — COMMENTED** [review]
LGTM, two small comments

**Pull request merged**